src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -103,9 +132,23 @@ public void launch() {
             kernel.getContext().get(c).startup();
         }
 
+        final List<DeviceIdentityInterface> provisioningPlugins = findProvisioningPlugins();
         // Must be called before everything else so that these are available to be
         // referenced by main/dependencies of main
         final Queue<String> autostart = findBuiltInServicesAndPlugins(); //NOPMD
+        loadPlugins();
+        // run the provisioning if device is not provisioned
+        if (!kernel.getContext().get(DeviceConfiguration.class).isDeviceConfiguredToTalkToCloud()
+                && !provisioningPlugins.isEmpty()) {
+            // Multiple provisioning plugins may need plugin ordering. We do not support plugin ordering right now
+            // There is also no compelling use case right now for multiple provisioning plugins.
+            if (provisioningPlugins.size() > 1) {
+                String errorString = String.format(MULTIPLE_PROVISIONING_PLUGINS_FOUND_EXCEPTION,
+                        provisioningPlugins.toString());
+                throw new RuntimeException(errorString);
```

Comment:
Plugins do not get loaded until loadPlugins is called. EZPLugins scan only once and they scan for built in services/plugins and provisioning plugin in one go only.

Reply:
Why does EZPlugins scan only once? even if that's a genuine concern, we should be able to change executeProvisioningPlugins to take a single plugin instead of a list

Reply:
I would think the scan is expensive, thats why once. Updated the execute method to take a single plugin.